src/lib/common/topology_sampler.cpp

```diff
@@ -64,53 +71,98 @@ static void topology_sample_cb(networkTopology_t* networkTopology) {
 
     // compile all additional info here
     _node_list.num_nodes = networkTopology->length;
-    configASSERT(sizeof(_node_list.nodes) >= _node_list.num_nodes * sizeof(uint64_t)); // if we hit this, expand the node list size
+    configASSERT(sizeof(_node_list.nodes) >=
+                 _node_list.num_nodes *
+                     sizeof(uint64_t)); // if we hit this, expand the node list size
 
-    network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.num_nodes), sizeof(_node_list.num_nodes));
+    network_crc32_calc = crc32_ieee_update(network_crc32_calc,
+                                           reinterpret_cast<uint8_t *>(&_node_list.num_nodes),
+                                           sizeof(_node_list.num_nodes));
 
     memset(_node_list.nodes, 0, sizeof(_node_list.nodes));
 
     neighborTableEntry_t *cursor = NULL;
     uint16_t counter;
-    for(cursor = networkTopology->front, counter = 0; (cursor != NULL) && (counter < _node_list.num_nodes); cursor = cursor->nextNode, counter++) {
+    xQueueReset(sys_info_crc_queue);
+    for (cursor = networkTopology->front, counter = 0;
+         (cursor != NULL) && (counter < _node_list.num_nodes);
+         cursor = cursor->nextNode, counter++) {
       _node_list.nodes[counter] = cursor->neighbor_table_reply->node_id;
-      network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.nodes[counter]), sizeof(uint64_t));
+      network_crc32_calc = crc32_ieee_update(
+          network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.nodes[counter]),
+          sizeof(uint64_t));
+      if (!sys_info_service_request(_node_list.nodes[counter], sys_info_reply_cb,
+                                    NODE_SYS_INFO_REQUEST_TIMEOUT_S)) {
+        printf("Failed to send sys info request to node: %" PRIu64 "\n",
+               _node_list.nodes[counter]);
+        break;
+      }
+    }
+
+    // Wait for all the sys info replies to come in
+    uint32_t node_crc_rx_count = 0;
+    while (node_crc_rx_count < _node_list.num_nodes) {
+      uint32_t node_crc = 0;
+      // Update the network crc with all of the node crcs
+      if (xQueueReceive(sys_info_crc_queue, &node_crc,
```

Comment:
We wait on all of the CRC's from the nodes to come back and aggregate them on the network crc.